Assets/Scripts/Models/Character/CharacterManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using MoonSharp.Interpreter;
+using UnityEngine;
+
+/// <summary>
+/// Character manager that holds all the characters.
+/// </summary>
+[MoonSharpUserData]
+public class CharacterManager
+{
+    private List<Character> characters;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CharacterManager"/> class.
+    /// </summary>
+    public CharacterManager()
+    {
+        characters = new List<Character>();
+    }
+
+    /// <summary>
+    /// Occurs when a character is created.
+    /// </summary>
+    public event Action<Character> CharacterCreated;
+
+    /// <summary>
+    /// Gets the characters.
+    /// </summary>
+    /// <value>An array of characters.</value>
+    public Character[] Characters
+    {
+        get { return characters.ToArray(); }
+    }
+
+    /// <summary>
+    /// Updates all the characters.
+    /// </summary>
+    /// <param name="deltaTime">Delta time.</param>
+    public void Update(float deltaTime)
+    {
+        // Change from a foreach due to the collection being modified while its being looped through
+        for (int i = characters.Count - 1; i >= 0; i--)
+        {
+            characters[i].Update(deltaTime);
+        }
+    }
+
+    /// <summary>
+    /// Create a Character in the specified tile.
+    /// </summary>
+    /// <param name="tile">The tile.</param>
+    public Character Create(Tile tile)
+    {
+        return Create(tile, ColorUtilities.RandomColor(), ColorUtilities.RandomGrayColor(), ColorUtilities.RandomSkinColor());
+    }
+
+    /// <summary>
+    /// Create a Character in the specified tile, with the specified color, uniform color and skin color.
+    /// </summary>
+    /// <param name="tile">The tile.</param>
+    /// <param name="color">The color.</param>
+    /// <param name="uniformColor">The  uniform color.</param>
+    /// <param name="skinColor">The  skin color.</param>
+    public Character Create(Tile tile, Color color, Color uniformColor, Color skinColor)
```

Comment:
Stylecop `SA1632 : CSharp.Documentation : The documentation text within the param tag must be at least 10 characters in length. Documentation failing to meet this guideline most likely does not follow a proper grammatical structure required for documentation text. (Assembly-CSharp)`